viewer/src/datamodels/cad/rendering/EffectRenderManager.ts

```diff
@@ -0,0 +1,132 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { MaterialManager } from '../MaterialManager';
+import { RenderMode } from './RenderMode';
+import * as THREE from 'three';
+import { Vector2, DataTexture } from 'three';
+import { edgeDetectionShaders } from './shaders';
+import { CogniteColors } from '@/utilities';
+
+export class EffectRenderManager {
+  private _materialManager: MaterialManager;
+  private _orthographicCamera: THREE.OrthographicCamera;
+  private _triScene: THREE.Scene;
+  private _windowTriangleMaterial: THREE.ShaderMaterial;
+  private _baseModelTarget: THREE.WebGLRenderTarget;
+  private _highlightedModelTarget: THREE.WebGLRenderTarget;
+
+  private readonly outlineTexelSize = 2;
+
+  constructor(materialManager: MaterialManager) {
+    this._materialManager = materialManager;
+    this._orthographicCamera = new THREE.OrthographicCamera(-1, 1, 1, -1, 0, 1);
+
+    this._triScene = new THREE.Scene();
+
+    const outlineColorTexture = this.createOutlineColorTexture();
+
+    this._windowTriangleMaterial = new THREE.ShaderMaterial({
+      vertexShader: edgeDetectionShaders.vertex,
+      fragmentShader: edgeDetectionShaders.fragment,
+      uniforms: {
+        tOutlineColors: { value: outlineColorTexture }
+      }
+    });
+    this.setupTextureRenderScene();
+
+    this._baseModelTarget = new THREE.WebGLRenderTarget(0, 0, { stencilBuffer: false });
+    this._highlightedModelTarget = new THREE.WebGLRenderTarget(0, 0, { stencilBuffer: false });

```

Comment:
I don't know what you mean by safe exactly, but the size is updated later as I don't know the size of the renderer yet.

Reply:
Yeah, noticed. I was just wondering if webgl might barf when it initializes a 0-size target. But my guess is that it's not initialized yet since there is no context involved.